docs/server-options.md

```diff
@@ -0,0 +1,116 @@
+---
+id: server-options
+title: Temporal server options
+sidebar_label: Start options
+---
+
+## Overview
```

Comment:
I think we should mention that by default you don't need any of these and the default docker build should be used. 

Only if you want to compile the service yourself to override certain components this approach should be used. And we want to list up front in which specific scenarios it can be useful.